scipy/signal/signaltools.py

```diff
@@ -501,17 +615,38 @@ def convolve(in1, in2, mode='full'):
         # Convolution is commutative; order doesn't have any effect on output
         volume, kernel = kernel, volume
 
+    if method == 'fft' and not _numeric_arrays([volume, kernel]):
+        raise ValueError('convolve can only be called with method=="fft" when'
+                         ' arrays consist of numeric elements (i.e., '
+                         'int/float/etc)')
+
+    if method == 'auto':
+        method = 'fft' if _fftconv_faster(volume, kernel, mode) else 'direct'
+        method = 'direct' if not _numeric_arrays([volume, kernel]) else method
+
+    if method in ('fft', 'auto'):
+        method = 'direct' if volume.dtype == 'complex256' else method
```

Comment:
testing for complex256 this way doesn't work if the computer doesn't support it.  

    not hasattr(np, "complex256") or a.dtype != np.complex256

was recommended: https://github.com/numpy/numpy/issues/5329#issuecomment-206698995